compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -698,6 +861,29 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
     }
 }
 
+pub(crate) fn push_lang_item_trait<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    bounds: &mut Vec<(ty::Clause<'tcx>, Span)>,
+    ty: Ty<'tcx>,
+    lang_item: LangItem,
+    span: Span,
+) {
+    assert_eq!(lang_item.target(), rustc_hir::Target::Trait);
+    if lang_item == LangItem::Sized {
+        let sized_def_id = tcx.require_lang_item(LangItem::Sized, Some(span));
+        let trait_ref = ty::TraitRef::new(tcx, sized_def_id, [ty]);
+        // Preferable to put this obligation first, since we report better errors for sized ambiguity.
+        bounds.insert(0, (trait_ref.upcast(tcx), span));
+    } else {
+        // Do not generate default bounds if lang item was not defined.
+        let Some(trait_def_id) = tcx.lang_items().get(lang_item) else {
```

Comment:
this can always require the lang item if you only call it in cases where `let trait_id = self.tcx().lang_items().get(trait_);` succeeded. You already checked whether the `DefId` exists to compute `do_not_provide_default_trait_bound`